services/automod/src/index.ts

```diff
@@ -0,0 +1,22 @@
+import type { DiscordEventsMap } from '@automoderator/common';
+import { decode, encode, Env } from '@automoderator/common';
+import { PubSubRedisBroker } from '@discordjs/brokers';
+import { GatewayDispatchEvents } from 'discord-api-types/v10';
+import Redis from 'ioredis';
+import { container } from 'tsyringe';
+import * as rawRunners from './runners';
+import type { Runner } from './struct/Runner';
+import { logger } from './util/logger';
+
+const env = container.resolve(Env);
+
+const redis = new Redis(env.redisUrl);
+const broker = new PubSubRedisBroker<DiscordEventsMap>({ redisClient: redis, encode, decode });
+
+broker.on(GatewayDispatchEvents.MessageCreate, async ({ data: message, ack }) => {
+	const runners: Runner[] = Object.values(rawRunners).map((runner) => container.resolve(runner));
+	// TODO: Run perms checks and the runners.
```

Comment:
Regarding this, don't try to re-build the old `PermissionsChecker` - have all the logic here, we can refactor it out in the future once its needed in multiple places.

Reply:
So, we are still gonna run perms checks here on this PR but without the `PermissionsChecker` and instead have all the perms logic here? thonk

Reply:
Yes. We'll refactor it into something generic later on. I'm just not sure what the PermissionsChecker currently needs while accounting for everything else.